modules/phase_field/src/action/MultiAuxVariablesAction.C

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MultiAuxVariablesAction.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+#include "libmesh/string_to_enum.h"
+
+template<>
+InputParameters validParams<MultiAuxVariablesAction>()
+{
+  MooseEnum familyEnum = AddAuxVariableAction::getAuxVariableFamilies();
+  MooseEnum orderEnum = AddAuxVariableAction::getAuxVariableOrders();
+
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up auxvariables for a polycrystal sample");
+  params.addParam<MooseEnum>("family", familyEnum, "Specifies the family of FE shape functions to use for this variable");
+  params.addParam<MooseEnum>("order", orderEnum, "Specifies the order of the FE shape function to use for this variable");
+  params.addRequiredParam<std::vector<unsigned int> >("op_num", "Vector that specifies the number of order parameters to create");
+  params.addRequiredParam<std::vector<std::string> >("var_name_base", "Vector that specifies the base name of the variables");
+  params.addParam<std::vector<SubdomainName> >("block", "The block id where this variable lives");
+  params.addParam<std::vector<OutputName> >("outputs", "Vector of output names where you would like to restrict the output of variable(s) associated with this object");
```

Comment:
the `initial_condition` parameter is a bit of an open sore now. If `AddVariableAction::createInitialConditionAction()` was virtual this action could overwrite it in an elegant way. Until then it might be best to

```
params.suppressParameter<Real>("initial_condition");
```